src/schema/posts.ts

```diff
@@ -2390,6 +2363,69 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         ),
       }));
     },
+    deleteSourcePostModeration: async (
+      _,
+      { postId }: { postId: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      const moderation = await ctx.con
+        .getRepository(SourcePostModeration)
+        .findOneByOrFail({ id: postId });
+
+      await ensureSourcePermissions(ctx, moderation.sourceId);
+
+      const isModerator = await isPrivilegedMember(ctx, moderation.sourceId);
```

Comment:
This reads from Source and SourceMember two times in a row, since you introduced `isPrivilegedMember` all over this maybe just use that one?

Reply:
Maybe we shouldn't have to check the moderator since they can reject the request. Let's just make author-only access to simplify.